File: fsharp/clock/Clock.fs
module Clock

type Clock = Clock of Minutes:int

let private modulo x y = (x % y + y) % y

let private minutesPerHour = 60
let private minutesPerDay = 1440

let create hours minutes = Clock (modulo (hours * minutesPerHour + minutes) minutesPerDay)

let add minutesToAdd (Clock minutes) = create 0 (minutes + minutesToAdd)

let subtract minutesToSubtract (Clock minutes) = create 0 (minutes - minutesToSubtract)

let display (Clock minutes) = $"%02i{minutes / minutesPerHour}:%02i{minutes % minutesPerHour}"

File: nim/clock/clock.nim
import math, strformat

type
  Clock = int
  Time = (int, int)

const MinutesPerHour = 60
const HoursPerDay = 24
const MinutesPerDay = HoursPerDay * MinutesPerHour

proc create*(time: Time): Clock =
  let (hours, minutes) = time
  floorMod(hours * MinutesPerHour + minutes, MinutesPerDay)

proc add*(time: Time, minutesToAdd: int): Clock =
  let (hours, minutes) = time
  create((hours, minutes + minutesToAdd))

proc subtract*(time: Time, minutesToSubtract: int): Clock =
  let (hours, minutes) = time
  create((hours, minutes - minutesToSubtract))

proc hours(clock: Clock): int =
  clock div MinutesPerHour

proc minutes(clock: Clock): int =
  clock mod MinutesPerHour

proc toStr*(clock: Clock): string =
  &"{clock.hours:02}:{clock.minutes:02}"
